components/clarity-lsp/src/common/backend.rs

```diff
@@ -279,14 +278,19 @@ pub enum LspRequestResponse {
     Hover(Option<Hover>),
 }
 
-pub fn process_request(command: LspRequest, editor_state: &EditorStateInput) -> LspRequestResponse {
+pub fn process_request(
+    command: LspRequest,
+    editor_state: &mut EditorStateInput,
```

Comment:
Question:
Until know, I tried to avoid mutating the state in LSP requests.
State mutation only happened in LSP notification.
So that requests don't lock the state.
Now I only need to edit the state in the Initialization request, but I considered implementing this logic in the native and vscode bridges instead of passing it as a &mut here (which opens the door to more mutation in the future). Any opinion about that?

Reply:
As discussed off band, introducing a `LspMutatingRequest` would be a good way out, introducing a path for both mutating an non mutating requests.